pynng/nng.py

```diff
@@ -908,3 +989,103 @@ def close(self):
         check_err(lib.nng_pipe_close(self.pipe))
         self._closed = True
 
+
+class Message:
+    """
+    Python interface for nng_msg.  See
+    https://nanomsg.github.io/nng/man/tip/nng_msg.5.html
+
+    Messages are immutable.
+
+    Warning:
+
+        Access to the message's underlying data buffer can be accessed with the
+        ``buffer`` attribute.  However, care must be taken not to send a message
+        while a reference to the buffer is still alive; if the buffer is used after
+        a message is sent, a segfault or data corruption will result.
+
+    """
+
+    def __init__(self, data, pipe=None):
+        self._pipe = pipe
```

Comment:
Maybe this should use the setter property?

Reply:
+1

Reply:
Updating this also showed me a bug in my code: I thought you could "unset" a message's pipe by passing NULL, but I never tested it, and it caused problems.  (I'm not sure why I ever thought that would be possible; `nng_msg_set_pipe` doesn't take a pointer, it takes a `nng_pipe`... oh well!)